tensorly/decomposition/_parafac2.py

```diff
@@ -95,6 +95,70 @@ def _project_tensor_slices(tensor_slices, projections):
     return tl.stack(slices)
 
 
+class _bro_thesis_line_search:
+    def __init__(self, norm_tensor, svd, verbose=False, nn_modes=None):
+        self.norm_tensor = norm_tensor
+        self.svd = svd
+        self.verbose = verbose
+        self.acc_pow = 2.0  # Extrapolate to the iteration^(1/acc_pow) ahead
+        self.acc_fail = 0  # How many times acceleration have failed
+        self.max_fail = 4  # Increase acc_pow with one after max_fail failure
+        self.nn_modes = nn_modes
```

Comment:
```suggestion
    def __init__(self, norm_tensor, svd, verbose=False, nn_modes=None, acc_pow=2.0, max_fail=4):
        self.norm_tensor = norm_tensor
        self.svd = svd
        self.verbose = verbose
        self.acc_pow = acc_pow  # Extrapolate to the iteration^(1/acc_pow) ahead
        self.max_fail = max_fail  # Increase acc_pow with one after max_fail failure
        self.acc_fail = 0 # How many times acceleration have failed
        self.nn_modes = nn_modes
```

Maybe we want to add the possibility of setting different starting parameters?